sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Body,
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Patch,
+  HttpStatus,
+} from "@nestjs/common";
+import { InstitutionService } from "../../services";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, Groups } from "../../auth/decorators";
+import { UserGroups } from "../../auth/user-groups.enum";
+import {
+  InstitutionProfileDto,
+  InstitutionReadOnlyDto,
+} from "./models/institution.dto";
+import BaseController from "../BaseController";
+import { ClientTypeBaseRoute } from "../../types";
+import { InstitutionControllerService } from "./institution.controller.service";
+import { ApiTags, ApiOperation, ApiResponse } from "@nestjs/swagger";
+/**
+ * Institution controller for AEST Client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("institution")
+@ApiTags("institution")
+export class InstitutionAESTController extends BaseController {
+  constructor(
+    private readonly institutionService: InstitutionService,
+    private readonly institutionControllerService: InstitutionControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get institution details of given institution.
+   * @param institutionId
+   * @returns InstitutionReadOnlyDto
+   */
+  @Get("/:institutionId")
+  @ApiOperation({
+    summary: "API for AEST client to get all institutions.",
+  })
+  @ApiResponse({ status: HttpStatus.ACCEPTED, description: "Accepted" })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: "Institution not valid.",
+  })
+  @ApiResponse({
```

Comment:
In general, I would say that we should add the _@ApiResponse_ for exceptions that we are generating in the controller.
For instance, NOT_FOUND, UNAUTHORIZED, BAD_REQUEST are pertinent to every single endpoint, so I would say that we should find a way to add them globally or avoid adding it to every single endpoint, unless it is generated from the controller body.

Reply:
Removed API level decorators as discussed until we find a generic way to do at controller level.